server/auth/sso/sso.go

```diff
@@ -323,16 +335,23 @@ func (s *sso) HandleCallback(w http.ResponseWriter, r *http.Request) {
 
 // authorize verifies a bearer token and pulls user information form the claims.
 func (s *sso) Authorize(authorization string) (*types.Claims, error) {
-	tok, err := jwt.ParseEncrypted(strings.TrimPrefix(authorization, Prefix))
+	bearerToken := authorization
+	if uncompressedToken, ok := s.bearerTokenMap.Get(authorization); ok {
+		bearerToken = uncompressedToken.(string)
+	}
+	tok, err := jwt.ParseEncrypted(strings.TrimPrefix(bearerToken, Prefix))
 	if err != nil {
+		s.bearerTokenMap.Remove(authorization)
 		return nil, fmt.Errorf("failed to parse encrypted token %v", err)
 	}
 	c := &types.Claims{}
 	if err := tok.Claims(s.privateKey, c); err != nil {
+		s.bearerTokenMap.Remove(authorization)
 		return nil, fmt.Errorf("failed to parse claims: %v", err)
 	}
 
 	if err := c.Validate(jwt.Expected{Issuer: issuer}); err != nil {
+		s.bearerTokenMap.Remove(authorization)
 		return nil, fmt.Errorf("failed to validate claims: %v", err)
 	}
```

Comment:
I'm curious - what's the user experience if the user's uuid got purged from the LRU cache and they are passing in their UUID? Will they see the error: "failed to parse encrypted token" or will they simply be asked to log in again?

Reply:
Yeah, it would say failed to parse.
And they get brought back to login page